build.gradle.kts

```diff
@@ -71,39 +71,31 @@ subprojects {
                 }
             }
 
-            val dependencies = configurations["compileClasspath"].resolvedConfiguration.resolvedArtifacts
-
-            dependencies.forEach { artifact ->
+            fun dependencyRules(artifact: ResolvedArtifact) {
                 val dependency = artifact.moduleVersion.id
-
                 if (dependency.group == project.group) {
                     val pathFromRoot = artifact.file.relativeTo(project.rootDir).path
                     val pathFromThisModule = artifact.file.relativeTo(project.projectDir).path
 
                     if (!dependency.name.endsWith("-spi") // modules may only depend on `-spi` modules (exceptions follow)
-                        && dependency.name != "spi" // exception: :spi module
-                        && !dependency.name.endsWith("-core") // exception: technology libs such as "blob-core"
-                        && !pathFromRoot.startsWith("common/") // exception: `common`
-                        && !pathFromRoot.startsWith("extensions/http/jetty/") // exception: `jetty` (this exception should be removed once there is an SPI for jetty)
-                        && !project.path.startsWith(":launchers:") // exception: launchers
-                        && !project.path.startsWith(":samples:") // exception: samples
-                        && !project.path.startsWith(":system-tests:") // exception: system-tests
+                            && dependency.name != "spi" // exception: :spi module
```

Comment:
I'm thinking if it would be useful to make the comments a bit more descriptive
Eg. instead of
> // exception: :spi module
// exception: modules may depend on `spi` module
// exception: modules may depend on `common` module
and then instead of
> // exception: launchers 
// exception: launchers may depend on other modules 
// exception: system-tests may depend on other modules  
etc

just to make it clear what are the rules. What do you think?

Reply:
The comments you refer to are exceptions to the rule mentioned in the first line:

``` // modules may only depend on `-spi` modules (exceptions follow)```

Basically this means that other `spi` modules,`launchers`, `samples`, etc are an exception to this rule and might depend on spi. IMO I think it is clear, but have another look and let me know if you have a suggestion.